.github/workflows/publish-docker.yaml

```
@@ -0,0 +1,40 @@
+name: "Publish"
+on:
+  workflow_dispatch:
+  # Run on PRs to main to make sure everything works, but we skip the publishing.
+  pull_request:
+    branches:
+      - main
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  publish:
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Checkout latest tag
+        run: |
+          git fetch --tags
+          TAG=$(git tag -l --sort=v:refname | tail -1)
+          echo "TAG=$TAG" >> $GITHUB_ENV
+          git checkout $TAG
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+      - uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: ${{ github.ref == 'refs/heads/main' }}
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ghcr.io/${{ github.repository_owner }}/pev2:${{ env.TAG }}
+            ghcr.io/${{ github.repository_owner }}/pev2:latest
```

Hey @alexklibisz I was trying to self-host the service and found this PR. I forked the repo and tried using the workflow to build the image for my usage but found an issue.

The tag is of the format `[HOST[:PORT_NUMBER]/]PATH` where the path is `[NAMESPACE/]REPOSITORY`.

In this case, the tag: `ghcr.io/dalibo/pev2:<tag>` is invalid. I tried to change it to `ghcr.io/dalibo/explain.dalibo.com/pev2:<tag>` which is also invalid.

See: https://github.com/docker/setup-buildx-action/discussions/291 & https://docs.docker.com/reference/cli/docker/image/tag/

---

This is what I ended up using to make it work.

```suggestion
            ghcr.io/${{ github.repository_owner }}/explain-dalibo-com/pev2:${{ env.TAG }}
            ghcr.io/${{ github.repository_owner }}/explain-dalibo-com/pev2:latest
```